yearn/test/system/saddle-strategy.test.js

```diff
@@ -0,0 +1,195 @@
+const { expect } = require("chai")
+const {
+  resetFork,
+  to1ePrecision,
+  impersonateAccount,
+  increaseTime,
+} = require("../helpers/contract-test-helpers.js")
+const { yearn, tbtc, forkBlockNumber } = require("./constants.js")
+const { deployYearnVault } = require("./functions.js")
+const { saddleStrategyFixture } = require("./fixtures.js")
+
+const describeFn =
+  process.env.NODE_ENV === "system-test" ? describe : describe.skip
+
+describeFn("System -- saddle strategy", () => {
+  let vaultGovernance
+  let vaultDepositor
+  let saddlePoolGovernance
+  let saddlePool
+  let tbtcSaddlePoolLPToken
+  let vault
+  let strategy
+
+  before(async () => {
+    await resetFork(forkBlockNumber)
+
+    // Setup roles.
+    vaultGovernance = await ethers.getSigner(0)
+
+    vaultDepositor = await impersonateAccount(
+      tbtc.saddlePoolLPTokenHolderAddress,
+      vaultGovernance
+    )
+
+    saddlePoolGovernance = await impersonateAccount(
+      tbtc.saddlePoolOwner,
+      vaultGovernance
+    )
+
+    // Get tBTC v2 Saddle pool handle and set `gated` to false to allow
+    // non-externally-owned accounts to perform staking
+    saddlePool = await ethers.getContractAt(
+      "ILPRewards",
+      tbtc.saddlePoolAddress
+    )
+    await saddlePool.connect(saddlePoolGovernance).setGated(false)
```

Comment:
This is not Saddle pool. This is Saddle LP rewards.

Reply:
Done in https://github.com/keep-network/tbtc-v2/commit/045e79a6b12147241e18295132a5992350505be2.